### PR TITLE
Release page show all tags in compare dropdown

### LINKED ISSUE #19936: Compare Releases 10 tags only
### Description

Pressing Compare button in Releases page, the 10 oldest releases only are shown to select for comparison.

### Gitea Version

1.16.8 ... main

### PR DESCRIPTION
**Pull request by 6543**
this fix #19936 but long term we should limit and use keyword to filter ...

### CODE REVIEW

File: routers/web/repo/release.go [hunk 1]
```diff
@@ -98,7 +98,12 @@ func releasesOrTags(ctx *context.Context, isTagList bool) {
 		listOptions.PageSize = setting.API.MaxResponseItems
 	}
 
-	tags, err := ctx.Repo.GitRepo.GetTags(listOptions.GetStartEnd())
+	tagListStart, tagListEnd := listOptions.GetStartEnd()
+	if !isTagList {
+		// tags are used for compare feature witch needs all tags
+		tagListStart, tagListEnd = 0, 0
+	}
```
Comment (Gusted) [inline routers/web/repo/release.go]: Reversing the logic, so you don't unnecessary call `GetStartEnd()`
```suggestion
	tagListStart, tagListEnd := 0, 0
	if isTagList {
		tagListStart, tagListEnd = listOptions.GetStartEnd()
	}
```

Comment (6543) [reply routers/web/repo/release.go]: this should not make a difference but I'll ajust ...
beside we should have it fixed later properly (#20073)

Comment (Gusted) [reply routers/web/repo/release.go]: > this should not make a difference but I'll ajust ...

Eh, seems it's better code-taste to use default values and then maybe the adjusted values. But feel free to ignore(really should prefix these comments with `code-style nits`)

Comment (6543) [reply routers/web/repo/release.go]: done

**Review by Gusted — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Comment by 6543** [comment]
@zeripath before the refactor (#16897) it was this way :D ...
anyway I created an issue #20073 to track this issue

**Pull request merged**